doc/TESTING_YOUR_CHANGES.md

```diff
@@ -0,0 +1,67 @@
+# Testing your changes and making sure they work as expected
+
+So you contributed or did something else that warrants testing. Nice work.
+Our tests will already cover a good amount of things that could go wrong,
+but you are expected to test your non-trivial changes in-game.
+Typo fixes and the like usually do not need to be tested.
+
+## Applying your changes locally
+
+A very simple way is to simply place your Cataclysm executable into your
+git repository.
+It will automagically use this altered path to load resources.
+For more advanced situations, there are commandline parameters such as
+`--datadir`, which you can point at your git respository.
+Check `--help` for more information.
+
+Exiting to the main menu and loading your save again will re-read most
+JSON files, so after making changes, make sure to either relaunch
+Cataclysm or reload your save, whatever is more convenient.
```

Comment:
```suggestion
If you need to test JSON-only changes, a very simple way is to simply place your Cataclysm executable into your git repository.
It will use this altered path to load resources.
For more advanced situations, there are commandline parameters such as
`--datadir`, which you can point at your git respository.
Check `--help` for more information.
You should also use an up-to-date version of Cataclysm and ensure that the git revision matches the Cataclysm executable version to avoid conflicts due to other people's changes to the JSON handling code.

Exiting to the main menu and loading your save again will re-read most
JSON files, so after making changes, make sure to either relaunch
Cataclysm or reload your save, whatever is more convenient.

For C++ changes, you'll need to compile the game, which is covered by `doc/COMPILING`.
```

Added some distinction between JSON and C++ changes. Simply placing the executable in the git repository has the potential of version mismatch so a note is also added.

For the altered path loading, it is a convention for most programs to use the resources in the directory they are located so there's nothing "automagical" about it really.